aws-redshift-cluster/src/test/java/software/amazon/redshift/cluster/UpdateHandlerTest.java

```diff
@@ -869,4 +851,79 @@ public void testModifyClusterAttributes(
         verify(proxyClient.client()).modifyCluster(any(ModifyClusterRequest.class));
     }
 
+    private static String NO_TEST_FLAG = "no-test-flag";
+    private static Stream<Arguments> maintenanceTrackTestData() {
+        return Stream.of(
+                Arguments.of("before-track", "after-track", true, NO_TEST_FLAG),
+                Arguments.of(null, "after-track", true, NO_TEST_FLAG),
+                Arguments.of("before-track", null, true, NO_TEST_FLAG),
+                Arguments.of("current", null, false, NO_TEST_FLAG),
+                Arguments.of(null, "current", false, NO_TEST_FLAG),
+                Arguments.of("current", "current", false, NO_TEST_FLAG),
+                Arguments.of(null, null, false, NO_TEST_FLAG),
+                Arguments.of("before-track", "after-track", true, "after-track"),
+                Arguments.of(null, "after-track", true, "after-track"),
+                Arguments.of("before-track", null, true, "current"),
+                Arguments.of("current", null, false, null),
+                Arguments.of(null, "current", false, null),
+                Arguments.of("current", "current", false, null),
+                Arguments.of(null, null, false, null)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("maintenanceTrackTestData")
+    public void testMaintenanceTrackModifyLogic(
+            String beforeTrack,
+            String afterTrack,
+            boolean shouldCallModifyCluster,
+            String expectedModifyTrackName
+    ) {
+        ResourceModel previousModel = BASIC_MODEL.toBuilder().publiclyAccessible(true).maintenanceTrackName(beforeTrack).build();
```

Comment:
Is this intended to have publicly Accessible clusters? if yes, why?

Reply:
in this unit test, I'm setting this to trigger the modify-cluster call. We need this extra unit test because in case we need to call modify-cluster but w/o any track changes, we need to pass in "null" to avoid modify-cluster failures